lumen_runtime/src/otp/maps/put_3.rs

```diff
@@ -16,57 +16,7 @@ use liblumen_alloc::erts::process::Process;
 use liblumen_alloc::erts::term::{Atom, Boxed, Map, Term};
 use liblumen_alloc::{badmap, ModuleFunctionArity};
 
-pub fn place_frame_with_arguments(
-    process: &Process,
-    placement: Placement,
-    key: Term,
-    value: Term,
-    map: Term,
-) -> Result<(), Alloc> {
-    process.stack_push(value)?;
-    process.stack_push(key)?;
-    process.stack_push(map)?;
-    process.place_frame(frame(), placement);
-
-    Ok(())
-}
-
-// Crate Public
-
-pub(in crate::otp) fn code(arc_process: &Arc<Process>) -> code::Result {
-    arc_process.reduce();
-
-    let map = arc_process.stack_pop().unwrap();
-    let key = arc_process.stack_pop().unwrap();
-    let value = arc_process.stack_pop().unwrap();
-
-    match native(arc_process, key, value, map) {
-        Ok(map) => {
-            arc_process.return_from_call(map)?;
-
-            Process::call_code(arc_process)
-        }
-        Err(exception) => result_from_exception(arc_process, exception),
-    }
-}
-
-// Private
-
-fn frame() -> Frame {
-    Frame::new(module_function_arity(), code)
-}
-
-fn function() -> Atom {
-    Atom::try_from_str("put").unwrap()
-}
-
-fn module_function_arity() -> Arc<ModuleFunctionArity> {
-    Arc::new(ModuleFunctionArity {
-        module: super::module(),
-        function: function(),
-        arity: 3,
-    })
-}
+bif!(super::module(), "put", native, key, value, map);
```

Comment:
According to cargo-expand, generates:

``` rust
    fn function() -> Atom {
        Atom::try_from_str("put").unwrap()
    }
    fn module_function_arity() -> Arc<ModuleFunctionArity> {
        Arc::new(ModuleFunctionArity {
            module: super::module(),
            function: function(),
            arity: 1usize + (1usize + (1usize + 0usize)),
        })
    }
    pub(in crate::otp) fn code(arc_process: &Arc<Process>) -> code::Result {
        arc_process.reduce();
        let key = arc_process.stack_pop().unwrap();
        let value = arc_process.stack_pop().unwrap();
        let map = arc_process.stack_pop().unwrap();
        match native(arc_process, key, value, map) {
            Ok(value) => {
                arc_process.return_from_call(value)?;
                Process::call_code(arc_process)
            }
            Err(exception) => result_from_exception(arc_process, exception),
        }
    }
    fn frame() -> Frame {
        Frame::new(module_function_arity(), code)
    }
    pub fn place_frame_with_arguments(
        process: &Process,
        placement: Placement,
        key: Term,
        value: Term,
        map: Term,
    ) -> Result<(), Alloc> {
        process.stack_push(map)?;
        process.stack_push(value)?;
        process.stack_push(key)?;
        process.place_frame(frame(), placement);
        Ok(())
    }
```

Reply:
My assumption is that `1usize + (1usize + (1usize + 0usize))` will get optimized by the compiler to `3`.